middle_end/flambda/types/env/typing_env_level.rec.ml

```diff
@@ -445,10 +462,28 @@ let construct_joined_level envs_with_levels ~env_at_fork ~allowed
       Variable.Map.empty
       envs_with_levels
   in
+  let depth_variables =
+    List.fold_left (fun depth_variables (_env_at_use, _id, _use_kind, t) ->
+        let dvs_this_level =
+          Depth_variable.Map.filter (fun dv _ ->
+              Typing_env.mem_depth_variable ~min_name_mode:Name_mode.normal
+                env_at_fork dv
+                  || Name_occurrences.mem_depth_variable allowed dv)
```

Comment:
I think this second disjunct is always going to be false at the moment; the definition of `allowed` below would need updating for this not to be the case.  I wonder if we could just remove this disjunct to start with.

Reply:
It comes in part from a parameter, no? I guess it depends on what the `extra_allowed_names` parameter to `join` is for?